### ISSUE TITLE
[discuss] stage 3 proposals, yay or nay?

### ISSUE THREAD

**Issue by bnoordhuis** [quickjs-ng/quickjs#670]
Did we ever settle on a policy on what to add when?

There's https://github.com/tc39/proposal-arraybuffer-base64 that I'd like to implement because:

1. it seems useful<sup>1</sup>
2. it is at stage 3
3. it is covered by test262 (feature `uint8array-base64`)
4. it is always nice to beat V8 on features :)

<sup>1</sup> Lets me remove the hand-rolled base64 parser I added in a1d1bce0b7a2682a890cbe6866e1cddbbb1ae940

**Comment by saghul** [comment]
I'm game 👍

**Comment by saghul** [comment]
I guess we could also use parse for atob?

**Comment by bnoordhuis** [comment]
If and when we implement that, yes :-)

I said in #16 I see no problem with adding DOMException but it kinda is slippery-sloping into web territory.

**Comment by saghul** [comment]
Right. We could have some "minimal web API helpers" thing perhaps?

Or we can just have those be implemented by embedders and that's that.

IIRC bjson already exports everything necessary for implementing structuredClone...

Shall we close #16?

**Comment by bnoordhuis** [comment]
I don't know, the argument that every embedder ends up implementing them anyway is still valid.

I'll work on arraybuffer-base64 and if afterwards atob turns out to be very easy to layer on top, then I guess we might as well do it. Given the choice, most users will probably pick more features over fewer features.

**Comment by saghul** [comment]
Good point.

**Comment by gengjiawen** [comment]
+1 from me. It's very useful. But maybe not atob and btoa. Very confuse name, really bad api design.
